pkgs/applications/networking/p2p/opentracker/default.nix

```diff
@@ -1,20 +1,75 @@
-{ lib, stdenv, fetchgit, libowfat, zlib, nixosTests }:
+{ lib
+, stdenv
+, fetchgit
+, libowfat
+, zlib
+, nixosTests
+, ipv6 ? false
+, blacklist ? false
+, whitelist ? false
+, syncLive ? false
+, ipFromQueryString ? false
+, restrictStats ? true
+, ipFromProxy ? false
+, fullLogNetwork ? false
+, logNumWant ? false
+, modestFullScrape ? fullscrape
+, spotWoodpecker ? false
+, fullscrape ? false
+}:
+
+assert !blacklist || !whitelist;
+
+let
+  features = {
+    V6 = ipv6;
+    ACCESSLIST_BLACK = blacklist;
+    ACCESSLIST_WHITE = whitelist;
+    SYNC_LIVE = syncLive;
+    IP_FROM_QUERY_STRING = ipFromQueryString;
+    COMPRESSION_GZIP = true;
+    COMPRESSION_GZIP_ALWAYS = false;
+    LOG_NETWORKS = false;
+    RESTRICT_STATS = restrictStats;
+    IP_FROM_PROXY = ipFromProxy;
+    FULLLOG_NETWORKS = fullLogNetwork;
+    LOG_NUMWANT = logNumWant;
+    MODEST_FULLSCRAPES = modestFullScrape;
+    SPOT_WOODPECKER = spotWoodpecker;
+    SYSLOGS = false;
+    DEV_RANDOM = false;
+    FULLSCRAPE = fullscrape;
+  };
```

Comment:
How about we allow just passing in a features attr instead of doing mapping for all the options?

Reply:
Maybe, but this way they are nicely handled by `.override`.